pkg/ensurance/cm/cpumanager/cpu_manager.go

```diff
@@ -631,11 +639,32 @@ func fromZoneListToTopologyResult(zones topologyapi.ZoneList) (TopologyResult, e
 			}
 			if numCPUs := zones[i].Resources.Capacity.Cpu().Value(); numCPUs > 0 {
 				tr[nodeID] = NodeInfo{
-					CPUs:            int(numCPUs),
-					NumReservedCPUs: int(zones[i].Resources.ReservedCPUNums),
+					CPUs:               int(numCPUs),
+					NumReservedCPUs:    int(zones[i].Resources.ReservedCPUNums),
+					ReservedSystemCPUs: getReservedSystemCPUs(attributes),
 				}
 			}
 		}
 	}
 	return tr, nil
 }
+
+func getReservedSystemCPUs(attributes map[string]string) cpuset.CPUSet {
+	reservedSystemCPUsStr, exist := attributes[topologyapi.ReservedSystemCPUsAttributes]
+	if exist {
+		reservedSystemCPUs, errParse := cpuset.Parse(reservedSystemCPUsStr)
+		if errParse != nil {
+			// invalid cpu list is provided, set reservedSystemCPUs to empty, so it won't overwrite kubeReserved/systemReserved
+			klog.Errorf("Invalid ReservedSystemCPUs \"%s\"", reservedSystemCPUsStr)
+		}
+		return reservedSystemCPUs
+	}
+	return cpuset.NewCPUSet()
+}
+
+func shouldExcludeReservedCPUs(pod *corev1.Pod) bool {
+	if pod == nil {
```

Comment:
nit: actually we don't need to check because all pods listed from kubelet must not be nil

Reply:
just a protection for panic...